src/record-error-archival/src/index.ts

```diff
@@ -0,0 +1,37 @@
+import { recordErrorArchival } from "./recordErrorArchival"
+import config from "./config"
+import DatabaseClient from "./DatabaseClient"
+import { AuditLogApiClient } from "shared"
+import { logger } from "shared"
+
+export default async function doRecordErrorArchival(): Promise<void> {
+  const db = new DatabaseClient(
+    config.dbHost,
+    config.dbUser,
+    config.dbPassword,
+    config.dbSsl,
+    config.dbName,
+    config.dbSchema,
+    config.archiveGroupLimit
+  )
+  const auditLogApi = new AuditLogApiClient(config.apiUrl, config.apiKey)
+
+  try {
+    const statuses = await recordErrorArchival(db, auditLogApi)
+    const successfulRecords = statuses?.filter((record) => record.success).length ?? 0
+    logger.info({
+      message: `Successfully updated ${successfulRecords} records`,
+      successfulRecords: successfulRecords
+    })
+
+    const failedRecords = statuses?.filter((record) => !record.success)
+
+    if (failedRecords !== undefined && failedRecords.length > 0) {
```

Comment:
We should be able to simplify this to
`if (failedRecords?.length > 0)`
or
`if (failedRecords && failedRecords.length > 0)`